test/WebJobs.Extensions.Tests/Extensions/Http/ClaimsPrincipalEndToEndTests.cs

```diff
@@ -0,0 +1,206 @@
+            _config = new JobHostConfiguration
+            {
+                TypeLocator = new ExplicitTypeLocator(typeof(TestFunctions))
+            };
+            _config.UseHttp(httpConfig);
+            _host = new JobHost(_config);
+        }
+
+        [Fact]
+        public void ClaimsPrincipalBindingWithEasyAuth()
+        {
+            var request = HttpTestHelpers.CreateHttpRequest("GET", "http://functions.com/api/TestIdentityBindings");
+            string easyAuthHeaderValue = Convert.ToBase64String(Encoding.UTF8.GetBytes(UserIdentityJson));
+            request.Headers.Add("x-ms-client-principal", easyAuthHeaderValue);
+
+            var method = typeof(TestFunctions).GetMethod(nameof(TestFunctions.HttpRequest));
+            _host.Call(method, new { req = request });
+
+            Assert.Equal(Name, globalPrincipal.Identity.Name);
+            Assert.Equal(UserAuthType, globalPrincipal.Identity.AuthenticationType);
+            Claim testClaim = globalPrincipal.Claims.Where(claim => string.Equals(claim.Type, UserClaimType)).First();
+            Assert.Equal(testClaim.Value, UserClaimValue);
+        }
+
+        [Fact]
+        public void ClaimsPrincipalBindingWithFunctionsKey()
+        {
+            var request = HttpTestHelpers.CreateHttpRequest("GET", "http://functions.com/api/TestIdentityBindings");
+            string keyHeaderValue = Convert.ToBase64String(Encoding.UTF8.GetBytes(KeyIdentityJson));
+            request.Headers.Add("x-ms-functions-key-identity", keyHeaderValue);
+
+            var method = typeof(TestFunctions).GetMethod(nameof(TestFunctions.HttpRequest));
+            _host.Call(method, new { req = request });
+
+            Assert.Equal(KeyAuthType, globalPrincipal.Identity.AuthenticationType);
+            Claim idClaim = globalPrincipal.Claims.Where(claim => string.Equals(claim.Type, KeyIdClaimType)).First();
+            Assert.Equal(idClaim.Value, KeyIdClaimValue);
+            Claim levelClaim = globalPrincipal.Claims.Where(claim => string.Equals(claim.Type, KeyLevelClaimType)).First();
+            Assert.Equal(levelClaim.Value, KeyLevelClaimValue);
+        }
+
+        [Fact]
+        public void CLaimsPrincipalBindingWithBothEasyAuthAndFunctionsKey()
+        {
+            var request = HttpTestHelpers.CreateHttpRequest("GET", "http://functions.com/api/TestIdentityBindings");
+            string easyAuthHeaderValue = Convert.ToBase64String(Encoding.UTF8.GetBytes(UserIdentityJson));
+            request.Headers.Add("x-ms-client-principal", easyAuthHeaderValue);
+            string keyHeaderValue = Convert.ToBase64String(Encoding.UTF8.GetBytes(KeyIdentityJson));
+            request.Headers.Add("x-ms-functions-key-identity", keyHeaderValue);
+
+            var method = typeof(TestFunctions).GetMethod(nameof(TestFunctions.HttpRequest));
+            _host.Call(method, new { req = request });
+
+            Assert.Equal(Name, globalPrincipal.Identity.Name);
```

Comment:
>Assert.Equal(Name, globalPrincipal.Identity.Name); [](start = 12, length = 50)

I suggest calling out in a comment that the user identity is expected to take precedence over the key identity.